astropy/io/fits/fitstime.py

```diff
@@ -175,22 +214,71 @@ def _verify_column_info(column_info, global_info):
             column_info['scale'] = global_info['scale']
             column_info['format'] = global_info.get('format', None)
 
-        # Non-linear coordinate types have "4-3" form and are not time coordinates
-        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+        else:
+            # Coordinate type is either an unrecognized local time scale
+            # or a linear coordinate type
             return False
 
-        # Coordinate type is either an unrecognized local time scale
-        # or a linear coordinate type
-        return False
+    elif (unit is not None and unit in FITS_TIME_UNIT) or location is not None:
```

Comment:
I'm not sure I quite understand when we get here: the column scale is undefined, but the unit is a time? isn't that a bit too greedy?